docs/Design/List.md

```diff
@@ -0,0 +1,117 @@
+# List
+
+## Feature Overview
+
+All supported languages provide some amount of built-in list support, either explicitly with lists or with an array equivalent.
+
+
+## Commands
+
+### `list pop`
+
+`list pop : name`
+
+The `list pop` command removes the last element of a list.
+
+## Usage
+
+```gls
+list pop : dogs

```

Comment:
We typically use `foo` in these. https://en.wikipedia.org/wiki/Metasyntactic_variable